sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/DiagnosticVerifier.cs

```diff
@@ -34,6 +38,42 @@
 
 namespace SonarAnalyzer.UnitTest.TestFramework
 {
+
+    class ErrorInUnitTest : Exception
+    {
+#pragma warning disable S3963 // "static" fields should be initialized inline
+        static ErrorInUnitTest()
+        {
+            var msCoreResourceManager = new ResourceManager("mscorlib", typeof(Object).Assembly);
+            at = msCoreResourceManager.GetString("Word_At");
+            inFileLineNum = msCoreResourceManager.GetString("StackTrace_InFileLineNumber");
+        }
+#pragma warning restore S3963 // "static" fields should be initialized inline
+
+        private static string GetFullPathOfRealSourceCode(string path) =>
+            Path.GetFullPath(Path.Combine(@"..\..\TestCases", path));
+
+        public ErrorInUnitTest(Location loc, string message) : base(message)
+        {
+            PathToRealSource = GetFullPathOfRealSourceCode(loc.GetLineSpan().Path);
+            Line = loc.GetLineNumberToReport();
+        }
+
+        public override string StackTrace
+        {
+            get
+            {
+                var addedLine = at + " Analyzed code " + string.Format(inFileLineNum, PathToRealSource, Line);
+                return addedLine + Environment.NewLine + base.StackTrace;
```

Comment:
String interpolation would be neater.
Why not calculate the string in the constructor and avoid storing some of the other fields?